firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FirebaseAppDistributionImpl.java

```diff
@@ -211,42 +214,39 @@ public Task<Void> signInTester() {
 
   @Override
   public void signOutTester() {
-    setCachedNewRelease(null);
+    cachedNewRelease.set(null);
     signInStorage.setSignInStatus(false);
   }
 
   @Override
   @NonNull
   public synchronized Task<AppDistributionRelease> checkForNewRelease() {
-    return checkForNewReleaseTaskCache.getOrCreateTask(
-        () -> {
-          if (!isTesterSignedIn()) {
-            return Tasks.forException(
-                new FirebaseAppDistributionException(
-                    "Tester is not signed in", AUTHENTICATION_FAILURE));
-          }
+    if (!isTesterSignedIn()) {
+      return Tasks.forException(
+          new FirebaseAppDistributionException("Tester is not signed in", AUTHENTICATION_FAILURE));
+    }
```

Comment:
I also pulled this out of the `getOrCreateTask` because it does not actually depend on whether there is a cached task.